app/src/main/java/dev/shreyaspatil/covid19notify/ui/main/adapter/StateAdapter.kt

```diff
@@ -14,6 +14,7 @@ import java.text.SimpleDateFormat
 
 class StateAdapter : ListAdapter<Details, StateAdapter.StateViewHolder>(DIFF_CALLBACK) {
 
+    var clickListener: (stateDetails: Details) -> Unit = { }
```

Comment:
You can try this and fork me the repo, I tried to do this and can't figure out, I 'll learn something new out of this, though passed the method in the constructor but couldn't figure out to do your way, But Both are correct, as far as I have heard it good practice to have a separate clicklistener method rather than having it in onBindView method or onBind method(standard Approach) though you can try it out and send me the fork.

Reply:
It's an opportunity for you to learn something new. Otherwise, this PR will be incomplete. Design is pretty good but the only design is not important, structuring business logic is more important. The motive behind this repo is not building the application but the main aim is to demonstrate best practices in Android development. So it's necessary to do it.